src/query/execute_schema.c

```diff
@@ -10016,6 +10020,366 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+  if (entity_name == NULL)
+    {
+      error = ER_UNEXPECTED;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, "Expecting a class or virtual class name.");
+      goto exit;
+    }
+
+  class_obj = db_find_class (entity_name);
+  if (class_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto exit;
+    }
+
+  error = locator_flush_class (class_obj);
+  if (error != NO_ERROR)
+    {
+      /* don't overwrite error */
+      goto exit;
+    }
+
+  /* force exclusive lock on class, even though it should have been already acquired */
+  if (locator_fetch_class (class_obj, DB_FETCH_WRITE) == NULL)
+    {
+      error = ER_FAILED;
+      goto exit;
+    }
+
+  ctemplate = dbt_edit_class (class_obj);
+  if (ctemplate == NULL)
+    {
+      /* when dbt_edit_class fails (e.g. because the server unilaterally aborts us), we must record the associated
+       * error message into the parser.	 Otherwise, we may get a confusing error msg of the form: "so_and_so is not a
+       * class". */
+      pt_record_error (parser, parser->statement_number - 1, alter->line_number, alter->column_number, er_msg (), NULL);
+      error = er_errid ();
+      goto exit;
+    }
+
+  error =
+    check_change_attribute (parser, ctemplate, alter->info.alter.alter_clause.attr_mthd.attr_def_list,
+                            alter->info.alter.alter_clause.attr_mthd.attr_old_name,
+                            &(alter->info.alter.constraint_list), &attr_chg_prop, &change_mode);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+
+  error = tran_system_savepoint (UNIQUE_SAVEPOINT_CHANGE_ATTR);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+  tran_saved = true;
+
+  error =
+    do_change_att_schema_only (parser, ctemplate, alter->info.alter.alter_clause.attr_mthd.attr_def_list,
+			      alter->info.alter.alter_clause.attr_mthd.attr_old_name,
+			      alter->info.alter.constraint_list, &attr_chg_prop, &change_mode);
+
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+
+  /* save class MOP */
+  class_mop = ctemplate->op;
+
+  /* check foreign key constraints */
+  error = do_check_fk_constraints (ctemplate, alter->info.alter.constraint_list);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+
+  is_srv_update_needed = ((change_mode == SM_ATTR_CHG_WITH_ROW_UPDATE || change_mode == SM_ATTR_CHG_BEST_EFFORT)
+			 && attr_chg_prop.name_space == ID_ATTRIBUTE) ? true : false;
+  if (is_srv_update_needed)
+    {
+      COPY_OID (&class_oid, &(ctemplate->op->oid_info.oid));
+      att_id = attr_chg_prop.att_id;
+    }
+
+  /* force schema update to server */
+  class_obj = dbt_finish_class (ctemplate);
+  if (class_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto exit;
+    }
+  /* set NULL, avoid 'abort_class' in case of error */
+  ctemplate = NULL;
+
+  if (attr_chg_prop.constr_info != NULL)
```

Comment:
constr_info is set at constraint altering, so I think that this conditional codes are not necessesary.

Reply:
Thank you. The code has been removed at the final commit.